--- LogKit-1.2.orig/src/java/org/apache/log/output/db/DefaultDataSource.java	2003-02-10 19:12:55.000000000 +0100
+++ LogKit-1.2/src/java/org/apache/log/output/db/DefaultDataSource.java	2008-10-23 19:46:49.000000000 +0200
@@ -149,4 +149,27 @@
     {
         m_logWriter = logWriter;
     }
+
+    /**
+     * @param iface
+     * @return
+     * @throws SQLException
+     * @see java.sql.Wrapper#isWrapperFor(java.lang.Class)
+     */
+    public boolean isWrapperFor(Class iface) throws SQLException {
+       throw new UnsupportedOperationException("This operation is not supported because we need to stay compatible " +
+                                               "with Java 1.4 where isWrapperFor() is not defined");
+    }
+
+    /**
+     * @param iface
+     * @return
+     * @throws SQLException
+     * @see java.sql.Wrapper#unwrap(java.lang.Class)
+     */
+    public Object unwrap(Class iface) throws SQLException {
+    //I hope that nothing will call this method (GK)
+        throw new UnsupportedOperationException("This operation is not supported because we need to stay compatible " +
+       		                                "with Java 1.4 where unwrap() is not defined");
+    }
 }
